src/librustc_mir/interpret/visitor.rs

```diff
@@ -208,6 +208,7 @@ macro_rules! make_value_visitor {
 
                 // Visit the fields of this value.
                 match v.layout().fields {
+                    FieldsShape::Primitive => {},
                     FieldsShape::Union(fields) => {
                         self.visit_union(v, fields)?;
```

Comment:
I think [this assertion](https://github.com/rust-lang/rust/blob/f0ae3979fe51c20d68cdd0291a298c014cf7f975/src/librustc_mir/interpret/validity.rs#L648) can now be simplified to just `assert!(fields > 0);`. Even better, we could remove the assertion there and have it here instead.

Reply:
Presumably the whole assertion is irrelevant if we switch to `NonZeroUsize`?

Reply:
Yes indeed. :)